cmd/agent/main.go

```diff
@@ -61,6 +63,22 @@ type GrpcProxyAgentOptions struct {
 	// Configuration for connecting to the proxy-server
 	proxyServerHost string
 	proxyServerPort int
+
+	agentID           string
+	syncInterval      int
```

Comment:
Not going to hold up merge but we should decide soon if we really want this to be int seconds or a duration.

Reply:
This field is going to hold the parsed flag value so it has to be int. I converted it to time.Duration when constructing the clients to simplify the code.

Reply:
I believe we do have support for parsing Duration flag fields as you can see here https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/options/options.go#L161.